src/main/scala/chisel3/util/circt/LTLIntrinsics.scala

```diff
@@ -0,0 +1,102 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.util.circt
+
+import chisel3._
+import chisel3.experimental.{IntParam, IntrinsicModule, Param, StringParam}
+import chisel3.experimental.hierarchy.{instantiable, public}
+
+import circt.Intrinsic
+
+private object Utils {
+  private[util] def withoutNone(params: Map[String, Option[Param]]): Map[String, Param] =
+    params.collect { case name -> Some(param) => name -> param }
+}
+
+@instantiable
+private[util] class BaseIntrinsic(
+  intrinsicName: String,
+  maybeParams:   Map[String, Option[Param]] = Map.empty[String, Option[Param]])
+    extends IntrinsicModule(f"circt_$intrinsicName", Utils.withoutNone(maybeParams)) {
+  private val paramValues = params.values.map(_ match {
+    case IntParam(x)    => x.toString
+    case StringParam(x) => x.toString
+    case x              => x.toString
+  })
+  override def desiredName = paramValues.fold(this.getClass.getSimpleName) { (a, b) => a + "_" + b }
+}
+
+/** Base class for all unary intrinsics with `in` and `out` ports. */
+@instantiable
+private[util] class UnaryLTLIntrinsic(
+  intrinsicName: String,
+  params:        Map[String, Option[Param]] = Map.empty[String, Option[Param]])
+    extends BaseIntrinsic(f"ltl_$intrinsicName", params) {
+  @public val in = IO(Input(Bool()))
+  @public val out = IO(Output(Bool()))
+}
+
+/** Base class for all binary intrinsics with `lhs`, `rhs`, and `out` ports. */
+@instantiable
+private[util] class BinaryLTLIntrinsic(
+  intrinsicName: String,
+  params:        Map[String, Option[Param]] = Map.empty[String, Option[Param]])
+    extends BaseIntrinsic(f"ltl_$intrinsicName", params) {
+  @public val lhs = IO(Input(Bool()))
+  @public val rhs = IO(Input(Bool()))
+  @public val out = IO(Output(Bool()))
```

Comment:
Generally `lhs` vs. `rhs` is for referring to the two sides of an assignment operator, not the two arguments to a binary function, I would not use that terminology for things like `and`.

Reply:
Yeah good point. These are part of the CIRCT-side interface of the intrinsics though, which I have to stick with here to emit the correct `intmodule`.